packages/ide-extension/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,148 @@
+            }
+        };
+    }
+    return reporter;
+}
+
+/**
+ * Update the telemetry setting by reading configuration.
+ *
+ * @returns - status of telemetry setting, true: enabled; false: disabled
+ */
+function updateTelemetryStatus(): boolean {
+    const enabled = !!workspace.getConfiguration('sap.ux.knowledgeHub').get('telemetryEnabled');
+    if (reporter) {
+        reporter.enabled = enabled;
+    }
+    return enabled;
+}
+
+/**
+ * Set common properties which will be added to every telemetry event.
+ * If called without properties, all common properties are removed.
+ *
+ * @param properties - name/value pair of properties (optional)
+ * @param properties.ide - development environment VSCODE or SBAS
+ * @param properties.devSpace - SBAS devspace
+ */
+export function setCommonProperties(properties?: { ide: 'VSCODE' | 'SBAS'; devSpace: string }) {
+    if (reporter) {
+        reporter.commonProperties = properties
+            ? {
+                  'cmn.appstudio': properties.ide === 'SBAS' ? 'true' : 'false',
+                  'cmn.devspace': properties.devSpace,
+                  apiHost: '',
+                  apiVersion: '',
+                  'common.os': platform(),
+                  'common.nodeArch': arch(),
+                  'common.platformversion': (release() || '').replace(/^(\d+)(\.\d+)?(\.\d+)?(.*)/, '$1$2$3'),
+                  'common.extname': packageJson.name,
+                  'common.extversion': packageJson.version
+              }
+            : undefined;
+    }
+}
+
+/**
+ * Track an even using telemetry.
+ *
+ * @param event - telemetry event
+ */
+export async function trackEvent(event: TelemetryEvent): Promise<void> {
+    if (!reporter?.enabled) {
+        return;
+    }
+    try {
+        const name = `${packageJson.name}/${event.name}`;
+        const properties = propertyValuesToString({ ...event.properties, ...(reporter.commonProperties || {}) });
+        reporter.client.trackEvent({ name, properties });
+        logString(`Telemetry event '${event.name}': ${JSON.stringify(event.properties)}`);
+    } catch (error) {
+        logString(`Error sending telemetry event '${event.name}': ${(error as Error).message}`);
+    }
+}
+
+/**
+ * Map specified redux actions to to telemetry events and track them.
+ *
+ * @param action - action that occurred
+ */
+export async function trackAction(action: any): Promise<void> {
+    if (!reporter?.enabled) {
+        return;
+    }
+    try {
+        const properties = {
+            action: '',
+            title: action.title,
+            primaryTag: action.primaryTag
+        };
+        if (action.source === 'tutorials') {
+            properties.action = 'OPEN_TUTORIALS';
+            trackEvent({ name: 'KHUB_OPEN_TUTORIAL', properties });
+        }
+        if (action.source === 'blogs') {
+            properties.action = 'OPEN_BLOGS';
+            trackEvent({ name: 'KHUB_OPEN_BLOGS', properties });
+        }
```

Comment:
Please use `else if` here or a switch case block. Also, use the action constants from the types module.